x-pack/legacy/plugins/uptime/public/components/connected/charts/monitor_duration.tsx

```diff
@@ -7,35 +7,80 @@
 import React, { useContext, useEffect } from 'react';
 import { useDispatch, useSelector } from 'react-redux';
 import { useUrlParams } from '../../../hooks';
-import { getMonitorDurationAction } from '../../../state/actions';
+import {
+  getAnomalyRecordsAction,
+  getMLCapabilitiesAction,
+  getMonitorDurationAction,
+} from '../../../state/actions';
 import { DurationChartComponent } from '../../functional/charts';
-import { selectDurationLines } from '../../../state/selectors';
+import {
+  anomaliesSelector,
+  hasMLFeatureAvailable,
+  hasMLJobSelector,
+  selectDurationLines,
+} from '../../../state/selectors';
 import { UptimeRefreshContext } from '../../../contexts';
+import { getMLJobId } from '../../../state/api/ml_anomaly';
+import { JobStat } from '../../../../../../../plugins/ml/common/types/anomaly_detection_jobs';
 
 interface Props {
   monitorId: string;
 }
 
 export const DurationChart: React.FC<Props> = ({ monitorId }: Props) => {
   const [getUrlParams] = useUrlParams();
-  const { dateRangeStart, dateRangeEnd } = getUrlParams();
+  const {
+    dateRangeStart,
+    dateRangeEnd,
+    absoluteDateRangeStart,
+    absoluteDateRangeEnd,
+  } = getUrlParams();
 
-  const { monitor_duration, loading } = useSelector(selectDurationLines);
+  const { durationLines, loading } = useSelector(selectDurationLines);
+
+  const isMLAvailable = useSelector(hasMLFeatureAvailable);
+
+  const { data: mlJobs } = useSelector(hasMLJobSelector);
+
+  const hasMLJob =
+    !!mlJobs?.jobsExist &&
+    !!mlJobs.jobs.find((job: JobStat) => job.id === getMLJobId(monitorId as string));
+
+  const anomalies = useSelector(anomaliesSelector);
 
   const dispatch = useDispatch();
 
   const { lastRefresh } = useContext(UptimeRefreshContext);
 
   useEffect(() => {
-    dispatch(
-      getMonitorDurationAction({ monitorId, dateStart: dateRangeStart, dateEnd: dateRangeEnd })
-    );
+    if (isMLAvailable) {
+      const anomalyParams = {
+        listOfMonitorIds: [monitorId],
+        dateStart: absoluteDateRangeStart,
+        dateEnd: absoluteDateRangeEnd,
+      };
+
+      dispatch(getAnomalyRecordsAction.get(anomalyParams));
+    }
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [dateRangeStart, dateRangeEnd, dispatch, lastRefresh, monitorId, isMLAvailable]);
```

Comment:
yeah i have added them dateRangeStart, dateRangeEnd to make sure date change is captured and can't add absoluteDateRangeStart, absoluteDateRangeEnd, because it will result in endless loop.